### PR TITLE
solution was added

### PR DESCRIPTION
**Pull request by Maksym845**
- [DEMO LINK](https://Maksym845.github.io/react_static-list-of-posts/)

### CODE REVIEW

File: src/components/PostInfo/PostInfo.tsx [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const { user, comments } = post;
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">
+          {post.title}
```
Comment (Yeliseiev) [inline src/components/PostInfo/PostInfo.tsx:19]:
```suggestion
  const { user, comments, title, body } = post;

  return (
    <div className="PostInfo">
      <div className="PostInfo__header">
        <h3 className="PostInfo__title">
          {title}
```

File: src/components/PostInfo/PostInfo.tsx [hunk 2]
```diff
@@ -1,5 +1,39 @@
...
+        </h3>
+
+        <UserInfo user={user} />
+      </div>
+
+      <p className="PostInfo__body">
+        {post.body}
```
Comment (Yeliseiev) [inline src/components/PostInfo/PostInfo.tsx:26]: Destructurize this too

File: src/components/PostInfo/PostInfo.tsx [hunk 3]
```diff
@@ -1,5 +1,39 @@
...
+      </p>
+      {comments.length !== 0
```
Comment (Yeliseiev) [inline src/components/PostInfo/PostInfo.tsx:28]:
```suggestion
      {comments.length
```

**Review by Yeliseiev — APPROVED** [review]
I think it is a good decision to transfer posts calculations to external functions, it is easier to read

File: src/App.tsx [hunk 4]
```diff
@@ -1,109 +1,45 @@
...
+function getUser(post: Post): User | null {
+  const user = usersFromServer.find(person => {
+    return person.id === post.userId;
+  });
+
+  if (user !== undefined) {
+    return user;
+  }
+
+  return null;
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:23]:
```suggestion

  return user || null;
```

Comment (OleksiiBlezniuk) [reply src/App.tsx:23]: This code can be shorter 🙂

File: src/components/CommentList/CommentList.tsx [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
+
+type Props = {
+  comments: Comment[];
+};
+
+export const CommentList: React.FC<Props> = ({ comments }) => (
+  <div className="CommentList">
+    {comments.map(comment => {
+      return <CommentInfo comment={comment} key={comment.id} />;
+    })}
```
Comment (OleksiiBlezniuk) [inline src/components/CommentList/CommentList.tsx:15]:
```suggestion
    {comments.map(comment => (
      <CommentInfo comment={comment} key={comment.id} />;
    ))}
```

Comment (OleksiiBlezniuk) [reply src/components/CommentList/CommentList.tsx:15]: Just return JSX after arrow, no need to use `return` keyword

File: src/components/PostList/PostList.tsx [hunk 6]
```diff
@@ -1,5 +1,18 @@
...
+};
+
+export const PostList: React.FC<Props> = ({ posts }) => (
+  <div className="PostList">
+    {posts.map(post => (
+      <PostInfo
+        post={post}
+        key={post.id}
+      />
+    ))}
```
Comment (OleksiiBlezniuk) [inline src/components/PostList/PostList.tsx:16]: Use semantic tags for lists and it's items

File: src/components/UserInfo/UserInfo.tsx [hunk 7]
```diff
@@ -1,5 +1,18 @@
...
+
+export const UserInfo: React.FC<Props> = ({ user }) => (
+  <p>
+    {` Posted by ${user?.username}: `}
+
+    <a className="UserInfo" href={`mailto:${user?.email}`}>
+      {user?.name}
+    </a>
+  </p>
 );
```
Comment (OleksiiBlezniuk) [inline src/components/UserInfo/UserInfo.tsx:18]: Don't render this component if user is null (add check in parent component)

File: src/components/CommentList/CommentList.tsx [hunk 8]
```diff
@@ -1,5 +1,17 @@
...
+type Props = {
+  comments: Comment[];
+};
+
+export const CommentList: React.FC<Props> = ({ comments }) => (
+  <div className="CommentList">
+    {comments.map(comment => (
+      <CommentInfo comment={comment} key={comment.id} />
+    ))}
+  </div>
```
Comment (Andr1yK) [inline src/components/CommentList/CommentList.tsx:16]:
```suggestion
  <ul className="CommentList">
    {comments.map(comment => (
      <CommentInfo comment={comment} key={comment.id} />
    ))}
  </ul>
```

**Review by Andr1yK — APPROVED** [review]

**Pull request closed without merging**